framework/src/postprocessors/ElementVectorL2Error.C

```diff
@@ -48,17 +71,14 @@ ElementVectorL2Error::getValue() const
 Real
 ElementVectorL2Error::computeQpIntegral()
 {
-  RealVectorValue sol_val(0.0, 0.0, 0.0);
-  RealVectorValue func_val(0.0, 0.0, 0.0);
-
-  sol_val(0) = _u[_qp];                          // required variable
-  func_val(0) = _funcx.value(_t, _q_point[_qp]); // required function
-
-  sol_val(1) = _v[_qp];
-  sol_val(2) = _w[_qp];
+  RealVectorValue func_val = isParamSetByUser("function")
```

Comment:
I would populate a `_has_vector_function` or something like that in the constructor so we don't have to peruse the parameter database every time we hit a quadrature point

Reply:
Good idea, done.